lib/src/main/java/uk/gov/di/data/lep/library/dto/deathnotification/DeathNotificationSetMapper.java

```diff
@@ -41,6 +44,66 @@ public static DeathNotificationSet generateDeathNotificationSet(GroJsonRecord gr
         );
     }
 
+    @Tracing
+    public static DeathNotificationSet generateMinimisedDeathNotificationSet(
+        DeathNotificationSet oldDeathNotificationSet,
+        List<EnrichmentField> enrichmentFields
+    ) {
+        if (oldDeathNotificationSet.events().deathRegistrationEvent() != null) {
+            var minimisedDeathRegistrationEvent = minimiseNewDeathRegistration(
+                oldDeathNotificationSet.events().deathRegistrationEvent(),
+                enrichmentFields
+            );
+            return new DeathNotificationSet(
+                oldDeathNotificationSet.aud(),
+                new DeathRegistrationEventMapping(minimisedDeathRegistrationEvent, null),
+                oldDeathNotificationSet.exp(),
+                oldDeathNotificationSet.iat(),
+                oldDeathNotificationSet.iss(),
+                oldDeathNotificationSet.jti(),
+                oldDeathNotificationSet.nbf(),
+                oldDeathNotificationSet.sub(),
+                oldDeathNotificationSet.toe(),
+                oldDeathNotificationSet.txn()
+            );
+        } else if (oldDeathNotificationSet.events().deathRegistrationUpdateEvent() != null) {
+            var minimisedDeathRegistrationUpdateEvent = minimiseUpdateDeathRegistration(
+                oldDeathNotificationSet.events().deathRegistrationUpdateEvent(),
+                enrichmentFields
+            );
+            return new DeathNotificationSet(
+                oldDeathNotificationSet.aud(),
+                new DeathRegistrationEventMapping(null, minimisedDeathRegistrationUpdateEvent),
+                oldDeathNotificationSet.exp(),
+                oldDeathNotificationSet.iat(),
+                oldDeathNotificationSet.iss(),
+                oldDeathNotificationSet.jti(),
+                oldDeathNotificationSet.nbf(),
+                oldDeathNotificationSet.sub(),
+                oldDeathNotificationSet.toe(),
+                oldDeathNotificationSet.txn()
+            );
+        }
+        throw new IllegalStateException("Both event types cannot be null");
+    }
+
+    @Tracing
+    private static DeathRegistrationUpdateEvent minimiseUpdateDeathRegistration(
+        DeathRegistrationUpdateEvent deathRegistrationUpdateEvent,
+        List<EnrichmentField> ignoredEnrichmentFields
+    ) {
+        return deathRegistrationUpdateEvent;
```

Comment:
Could we raise a ticket to implement this?  I'm assuming because DWP are just getting everything, it's not been necessary?

Reply:
GPC-536